File: app/src/main/java/com/ichizin/hatezin/ui/fragments/BaseFragment.java
package com.ichizin.hatezin.ui.fragments;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.View;

import com.ichizin.hatezin.util.Navigator;

import javax.inject.Inject;

import butterknife.ButterKnife;

/**
 *
 *
 * @author ichizin
 */
public class BaseFragment extends Fragment {

    @Inject
    Navigator navigator;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        ButterKnife.bind(this, view);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        ButterKnife.unbind(this);
    }
}

File: app/src/main/java/com/ichizin/hatezin/ui/fragments/BaseFragment.kt
package com.ichizin.hatezin.ui.fragments

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.View

import butterknife.ButterKnife

/**
 * BaseFragment
 * @author ichizin
 */
abstract class BaseFragment : Fragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        ButterKnife.bind(this, view)
    }

    override fun onResume() {
        super.onResume()
    }

    override fun onPause() {
        super.onPause()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        ButterKnife.unbind(this)
    }

    override fun onDestroy() {
        super.onDestroy()
    }
}
